### PR TITLE
Delete only users with a creation date

### PR DESCRIPTION
**Pull request by jens-hammel**

### CODE REVIEW

File: remove-user-on-logout/pom.xml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
```
Comment (ingokuba) [inline remove-user-on-logout/pom.xml:2]: I reformatted this file - make sure to fix your IntelliJ config:
image

File: remove-user-on-logout/src/test/java/de/intension/resources/admin/VidisAdminRealmResourceProviderIT.java [hunk 2]
```diff
@@ -63,11 +64,15 @@ class VidisAdminRealmResourceProviderIT
     public static final String               ADMIN_PASSWORD    = keycloak.getAdminPassword();
 
     @Test
-    void shouldDeleteAllUsers_whenAllConfigured()
+    void shouldDeleteUsersWithCreatedTimestam_whenAllConfigured()
```
Comment (ingokuba) [inline remove-user-on-logout/src/test/java/de/intension/resources/admin/VidisAdminRealmResourceProviderIT.java:67]: nitpick: *Timestamp

File: remove-user-on-logout/src/main/java/de/intension/resources/admin/VidisAdminRealmResourceProvider.java [hunk 3]
```diff
@@ -27,8 +27,9 @@ public class VidisAdminRealmResourceProvider
     implements AdminRealmResourceProvider
 {
 
-    private static final Logger      LOG                                       = Logger.getLogger(VidisAdminRealmResourceProvider.class);
-    private static final int         TOLERANCE_FOR_USER_IN_CREATION_IN_SECONDS = 5;
+    private static final Logger      LOG                                               = Logger.getLogger(VidisAdminRealmResourceProvider.class);
+    private static final int         DEFAULT_TOLERANCE_FOR_USER_IN_CREATION_IN_SECONDS = 30;
+    public static final String       DELETION_TOLERANCE_CONFIG                         = "deletiontolerance";
```
Comment (ingokuba) [inline remove-user-on-logout/src/main/java/de/intension/resources/admin/VidisAdminRealmResourceProvider.java:32]: shouldn't this new config param be documented? It's probably only configurable with system property / environment variable (`KC_SPI_...` right?)

File: remove-user-on-logout/src/main/java/de/intension/resources/admin/VidisAdminRealmResourceProvider.java [hunk 4]
```diff
@@ -108,8 +110,7 @@ private List<UserEntity> getListOfUsers(int chunkSize, long lastCreationDate, bo
                 : " ";
         Query userQuery = em.createNativeQuery("select ue.* "
                 + "from user_entity ue "
-                + "where (ue.created_timestamp > :lastTimeStamp "
-                + "or ue.created_timestamp is null) "
+                + "where ue.created_timestamp > :lastTimeStamp "
```
Comment (ingokuba) [inline remove-user-on-logout/src/main/java/de/intension/resources/admin/VidisAdminRealmResourceProvider.java:113]: so as I understand this is the actual change described by the PR title?
the new config param is additional improvment?

Comment (jens-hammel) [reply remove-user-on-logout/src/main/java/de/intension/resources/admin/VidisAdminRealmResourceProvider.java:113]: No the default tolerance time is also increased. So these are three changes. Don't delete users without created_timestamp. Ignore users younger then 30s (default). If that is to long it is configurable without needing to modify the code.

File: remove-user-on-logout/src/main/java/de/intension/resources/admin/VidisAdminRealmResourceProvider.java (hunk 3 above)
Comment (jens-hammel) [reply remove-user-on-logout/src/main/java/de/intension/resources/admin/VidisAdminRealmResourceProvider.java:32]: right will do.

File: remove-user-on-logout/pom.xml (hunk 1 above)
Comment (jens-hammel) [reply remove-user-on-logout/pom.xml:2]: @ingokuba Can you add that to our eclipse profile?
Reformatted it but that won't change that line. Continuation indent is 8...

**Pull request merged**